fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/adapters/EventExpandableListViewAdapter.java

```diff
@@ -242,15 +248,57 @@ public View getChildView(
     // in the future it could be nice to have a pencil icon to allow organizer to modify an event
 
     LayoutEventBinding binding;
-    Event event = ((Event) getChild(groupPosition, childPosition));
+
     if (convertView == null) {
       LayoutInflater inflater = LayoutInflater.from(parent.getContext());
       binding = LayoutEventBinding.inflate(inflater, parent, false);
     } else {
       binding = DataBindingUtil.getBinding(convertView);
     }
 
-    binding.setEvent(event);
+    //Event event = ((Event) getChild(groupPosition, childPosition));
+    Event event = (Event)getChild(groupPosition, childPosition);
+    if(event instanceof RollCall){
```

Comment:
I created an attribute EventType type for each event in the class Event so I think it would be nicer to do a switch case on this instead of having to do multiples if ( instanceof) else (instanceof) etc.. especially when there will be multiple events to handle.

Reply:
Honestly I don't think this is an issue and the lines of code stay the same anyway. I would prefer staying consistent with the style used overall in the code (for example in LAORepository for handleMessage) and this would also avoid the creation of an additional attribute.